### PR TITLE
125: Add filtering to get posts REST endpoint

### LINKED ISSUE #125: Add filtering to get posts REST endpoint
We need to add filtration to the `get_posts` REST route. Filtering should be possible by the following attributes: `author_id`, `created_before`, `created_after`.

**Request example**
```http
GET /topics/42/posts?author_id=42&author_id=24&author_id=18
```

**Response example**
```python
[
    # posts with author_id 42 or 24 or 18
]
```

We suppose that all datetime related query params will be sent in UTC timezone in the following format:
```text
'2023-04-24 05:10:20.782336'
```

**Request example**

```http
GET /topics/42/posts?created_after=2023-04-24 04:10:20.782336&created_before=2023-04-24 05:10:20.782336
```

So in the scope of this task we need to add filtration for `get_posts` route.

### PR DESCRIPTION
**Pull request by EugeneTurkin**
We need to add filtration to the `get_posts` REST route. Filtering should be possible by the following attributes: `author_id`, `created_before`, `created_after`.

**Request example**
```http
GET /topics/42/posts?author_id=42&author_id=24&author_id=18
```

**Response example**
```python
[
    # posts with author_id 42 or 24 or 18
]
```

We suppose that all datetime related query params will be sent in UTC timezone in the following format:
```text
'2023-04-24 05:10:20.782336'
```

**Request example**

```http
GET /topics/42/posts?created_after=2023-04-24 04:10:20.782336&created_before=2023-04-24 05:10:20.782336
```

So in the scope of this task we need to add filtration for `get_posts` route.

### CODE REVIEW

File: src/api/topics/routes.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ def get() -> list[dict[str, Any]] | None:
         parsed_args = parser.parse_args()
         topics = Topic.get_topics(
             sorting=parsed_args.get("order_by"),
-            author_ids=parsed_args.get("author_ids"),
+            author_ids=parsed_args.get("author_id"),
```
Comment (birthdaysgift) [inline src/api/topics/routes.py]: Why is this line changed? Doesn't it work with `.get("author_ids"` ?

**Review by birthdaysgift — CHANGES_REQUESTED** [review]

File: src/api/topics/routes.py (hunk 1 above)
Comment (EugeneTurkin) [reply src/api/topics/routes.py]: yes, it doesn't. we have an argument "author_id" defined in parser, but we try to get "author_ids" from parsed_args.

Comment (birthdaysgift) [reply src/api/topics/routes.py]: Ok, seems like it's a bug, so I created a new ticket for that: https://github.com/fenya123/forum123/issues/144

**Review by birthdaysgift — APPROVED** [review]
